zetaclient/chains/ton/observer/inbound.go

```diff
@@ -0,0 +1,260 @@
+package observer
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+	"slices"
+
+	"cosmossdk.io/math"
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog"
+	"github.com/tonkeeper/tongo/ton"
+
+	"github.com/zeta-chain/node/pkg/coin"
+	toncontracts "github.com/zeta-chain/node/pkg/contracts/ton"
+	"github.com/zeta-chain/node/pkg/ticker"
+	"github.com/zeta-chain/node/zetaclient/chains/ton/liteapi"
+	zctx "github.com/zeta-chain/node/zetaclient/context"
+	"github.com/zeta-chain/node/zetaclient/zetacore"
+)
+
+const (
+	// MaxTransactionsPerTick is the maximum number of transactions to process on a ticker
+	MaxTransactionsPerTick = 100
```

Comment:
Curious , whats the reason behind selecting this number

Reply:
The shortest tick is `1 sec`. I think none of our chain observers could process 100 inbounds in one ticker. The hardcoded `100` seems to me some sort of `estimated max inbound processing speed`. The speed of observation can be easily measured as it also depends on the performance of RPC endpoints.